libtsuba/include/tsuba/tsuba.h

```diff
@@ -87,6 +86,9 @@ struct KATANA_EXPORT RDGStat {
 /// Get Information about the graph
 KATANA_EXPORT katana::Result<RDGStat> Stat(const std::string& rdg_name);
 
+KATANA_EXPORT katana::Result<std::vector<RDGView>> ListAvailableViews(
+    const std::string& rdg_dir);
+
```

Comment:
I don't want to overdesign things, but if you look at the interfaces for listing pseudo-directories in buckets, they all return a "page" of results and a token for more.  I wonder if a limit parameter would be useful here to reduce memory use for large directories.  But then you'd need a start string (list everything equal to or later than the start string up to the given limit).

Reply:
Probably in the future this will be pretty useful. For now, I say we stick with this as it gets us off the ground.

Reply:
Pagination here seems like overkill to me as well. I think the number of views will be on the order of 10s or maybe 100s per version. (This does only return the view for the latest version right?)

Reply:
I actually missed the most important thing here which is that this is listing views of a version.  
Along the lines of OpaqueIDs preventing accidental cross-confusion, making views a typed enum will prevent auto casting from version numbers to view types (though now one is an integer and the other a string, so no present danger).

Reply:
Yeah, eventually once we start to see what kinds of views we support (aside from partitioned RDG views) we will create enums and probably create some kind of plugin-based view handler support. For now, this method simply returns all the views available in the directory to the user.